kyo-prelude/shared/src/main/scala/kyo/Emit.scala

```diff
@@ -109,21 +109,22 @@ object Emit:
           * @param acc
           *   The initial accumulator value
           * @param f
-          *   The folding function
+          *   The folding function that takes the current accumulator and emitted value, and returns a tuple of the new accumulator and an Ack
+          *   to control further emissions
           * @param v
           *   The computation with Emit effect
           * @return
           *   A tuple of the final accumulator value and the result of the computation
           */
-        def apply[A, S, B: Flat, S2](acc: A)(f: (A, V) => A < S)(v: B < (Emit[V] & S2))(
+        def apply[A, S, B: Flat, S2](acc: A)(f: (A, V) => (A, Ack) < S)(v: B < (Emit[V] & S2))(
```

Comment:
I've updated this method to be more generic and allow the function to determine if the stream should continue.